crates/mempool/src/estimate.rs

```diff
@@ -0,0 +1,272 @@
+use const_hex::hex;
+use core::fmt::Debug;
+use ethers::{
+    abi::{Hash, RawLog},
+    contract::EthLogDecode,
+    providers::Middleware,
+    types::{Bytes, U256},
+};
+use silius_contracts::{
+    decode_revert_string,
+    executor_tracer::{ExecutorTracerResult, LogInfo},
+    EntryPoint, EntryPointError, ExecutionResult, FailedOp, UserOperationEventFilter,
+    UserOperationRevertReasonFilter,
+};
+use silius_primitives::UserOperationSigned;
+use std::str::FromStr;
+
+const FALL_BACK_BINARY_SEARCH_CUT_OFF: u64 = 30000;
+const BASE_VGL_BUFFER: u64 = 25;
+const MAX_CALL_GAS_LIMIT: u64 = 18_000_000;
+const MAX_RETRY: u64 = 7;
+const NON_ZERO_GAS: u64 = 12100; // should be different based on diferrent chain
```

Comment:
Is this value chain specific? Then we should move this value to the chain struct and set different values for different chains

Reply:
We probably need chain spec thing to handle different values on different chains

Reply:
I am not sure what other chains the value is. It may be different or not.It is good to have it what it is right now.

Reply:
Ok, got it. But we should still handle the Optimism and Arbitrum differently - L1 gas and L2 gas separately (as other bundlers are handling)

Reply:
Sorry. We could not do it easily right now. It is another big task. Would you mind merging this first ? Created an issue for that https://github.com/silius-rs/silius/issues/290.